pkg/controller/job/job_controller.go

```diff
@@ -597,6 +599,23 @@ func (jm *JobController) deleteJobPods(job *batch.Job, pods []*v1.Pod, errCh cha
 	wait.Wait()
 }
 
+// pastBackoffLimit checks if container restartCounts sum exceeds BackoffLimit
+// this method applies only to pods with restartPolicy == OnFailure
+func pastBackoffLimit(job *batch.Job, pods []*v1.Pod) bool {
+	result := int32(0)
+	for i := range pods {
+		po := pods[i]
+		if po.Status.Phase != v1.PodRunning {
+			continue
+		}
+		for j := range po.Status.ContainerStatuses {
+			stat := po.Status.ContainerStatuses[j]
+			result += stat.RestartCount
```

Comment:
From `ContainerStatus` spec, this number is calculated from the number of dead containers and will be capped at 5 by GC. If the BackoffLimit is set to > 5 * (# of containers), the job may never past backoff limit. 

```go
	// The number of times the container has been restarted, currently based on
	// the number of dead containers that have not yet been removed.
	// Note that this is calculated from dead containers. But those containers are subject to
	// garbage collection. This value will get capped at 5 by GC.
	RestartCount int32 
```

Reply:
Agree, but I'm not sure I'd want to address this with some hacky approach in the controller. This is a thing a user should be aware of, more or less.